engines/leveldb/iter.go

```
@@ -2,58 +2,90 @@ package leveldb
 
 import (
 	"bytes"
-	"fmt"
 
 	honuiter "github.com/rotationalio/honu/iterator"
 	pb "github.com/rotationalio/honu/object"
+	opts "github.com/rotationalio/honu/options"
 	"github.com/syndtr/goleveldb/leveldb/iterator"
 	"google.golang.org/protobuf/proto"
 )
 
 // NewLevelDBIterator creates a new iterator that wraps a leveldb Iterator with object
 // management access and Honu-specific serialization.
-func NewLevelDBIterator(iter iterator.Iterator, namespace string) honuiter.Iterator {
-	return &ldbIterator{ldb: iter, namespace: namespace}
+func NewLevelDBIterator(iter iterator.Iterator, options *opts.Options) honuiter.Iterator {
+	return &ldbIterator{ldb: iter, options: options}
 }
 
 // Wraps the underlying leveldb iterator to provide object management access.
 type ldbIterator struct {
-	ldb       iterator.Iterator
-	namespace string
+	ldb     iterator.Iterator
+	options *opts.Options
 }
 
 // Type check for the ldbIterator
 var _ honuiter.Iterator = &ldbIterator{}
 
-func (i *ldbIterator) Next() bool   { return i.ldb.Next() }
-func (i *ldbIterator) Prev() bool   { return i.ldb.Prev() }
 func (i *ldbIterator) Error() error { return i.ldb.Error() }
 func (i *ldbIterator) Release()     { i.ldb.Release() }
 
+func (i *ldbIterator) Next() bool {
+	if ok := i.ldb.Next(); !ok {
+		return false
+	}
+
+	// If we aren't including Tombstones, we need to check if the next version is a
+	// tombstone before we know if we have a next value or not.
+	if !i.options.Tombstones {
+		if obj, err := i.Object(); err != nil || obj.Tombstone() {
```

Are we ignoring the error here or does it get captured somewhere else?